cargo-dist/src/linkage.rs

```diff
@@ -81,7 +82,21 @@ fn compute_linkage_assuming_local_build(
                 if !bin_path.exists() {
                     eprintln!("Binary {bin_path} missing; skipping check");
                 } else {
-                    let linkage = determine_linkage(&bin_path, target)?;
+                    let linkage = match determine_linkage(&bin_path, target) {
+                        Ok(linkage) => linkage,
+                        // We don't want to treat the primary linkage
+                        // failure check as a failure - instead, print
+                        // a warning and provide a default.
+                        Err(DistError::LinkageCheckUnsupportedBinary { file_name }) => {
+                            if let Some(file_name) = file_name {
+                                warn!("Unable to fetch linkage from {file_name}");
+                            } else {
+                                warn!("Unable to fetch linkage");
+                            }
+                            Linkage::default()
+                        }
+                        Err(e) => return Err(e),
```

Comment:
Why should we allow any linkage errors to kill anything?

Reply:
Good point, there are a handful of other errors here that we should probably just warn about instead of making nonfatal.

Reply:
Turned this into a "warn and move on" case too, just with a more generic warning than `LinkageCheckUnsupportedBinary`.